manifests/agent/facter.pp

```diff
@@ -0,0 +1,67 @@
+# Puppet agent facter configuration
+# @api private
+class puppet::agent::facter (
+  Optional[Array[String]] $blocklist = undef,
+  Optional[Array[String]] $cachelist = undef,
+  String $cache_ttl = '1 day',
+) {
+
+    if versioncmp(fact('aio_agent_version'),'7') >= 0 {
+    file { '/etc/puppetlabs/facter':
+      ensure => directory,
+    }
+
+        hocon_setting { 'facter.confi':
+      path    => '/etc/puppetlabs/facter/facter.conf',
+      require => File['/etc/puppetlabs/facter'],
+    }
+
+
+        if $::blocklist {
+      hocon_setting { 'blocklist facts group':
```

Comment:
can't we also write json to the facter conf? Then we don't need a dependency to the hocon module.